### PR TITLE
feat: improve restore

### PR DESCRIPTION
**Pull request by NikoManiac**
closes #

### CODE REVIEW

File: packages/backup-format/src/container/index.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { concatArrayBuffer, unreachable } from '@dimensiondev/kit'
 import { BackupErrors } from '../BackupErrors'
 
 const MAGIC_HEADER_Version0 = new Uint8Array([...'MASK-BACKUP-V000'].map((x) => x.charCodeAt(0)))
```
Comment (septs) [inline packages/backup-format/src/container/index.ts:4]:
```suggestion
const MAGIC_HEADER_Version0 = new TextEncoder().encode('MASK-BACKUP-V000')
```

File: packages/maskbook/src/extension/background-script/IdentityService.ts [hunk 2]
```diff
@@ -130,6 +133,12 @@ export function queryMyPersonas(network?: string): Promise<Persona[]> {
             : x,
     )
 }
+
+export async function queryLastPersonaCreated() {
+    const all = await queryPersonas(undefined, true)
+    return orderBy(all, (x) => x.createdAt, 'desc')[0]
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/background-script/IdentityService.ts:139]:
```suggestion
    return first(orderBy(all, (x) => x.createdAt, 'desc'))
```

File: packages/dashboard/src/components/Restore/RestoreFromCloud.tsx [hunk 3]
```diff
@@ -38,30 +54,59 @@ export const RestoreFromCloud = memo(() => {
...
+                    if (lastedPersona) {
+                        await changeCurrentPersona(lastedPersona.identifier)
+                    }
+                }
+                if (account) {
+                    if (!user.email && account.type === AccountType.email) {
+                        updateUser({ email: account.value })
+                    }
+                    if (!user.phone && account.type === AccountType.phone) {
+                        updateUser({ phone: account.value })
```
Comment (yanzhihong23) [inline packages/dashboard/src/components/Restore/RestoreFromCloud.tsx:100]: phone number: ``${countryCode} ${phone}``

Comment (septs) [reply packages/dashboard/src/components/Restore/RestoreFromCloud.tsx:100]: @yanzhihong23 plz read E.164 standard

**Review by guanbinrui — DISMISSED** [review]

**Comment by Jack-Works** [comment]
Please wait for my review, don't merge too fast thanks

File: packages/dashboard/src/components/Restore/RestoreFromCloud.tsx (hunk 3 above)
Comment (yanzhihong23) [reply packages/dashboard/src/components/Restore/RestoreFromCloud.tsx:100]: @septs not talking standard here. This value will be saved in localStorage. Just for separating country code and number.

**Review by guanbinrui — APPROVED** [review]

**Review by yanzhihong23 — APPROVED** [review]

**Review by Jack-Works — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**